analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSyntaxHelper.cs

```diff
@@ -282,8 +282,23 @@ node switch
         public static bool HasConstantValue(this ExpressionSyntax expression, SemanticModel semanticModel) =>
             expression.RemoveParentheses().IsAnyKind(LiteralSyntaxKinds) || expression.FindConstantValue(semanticModel) != null;
 
-        public static string GetStringValue(this SyntaxNode node, SemanticModel semanticModel) =>
-            CsharpStringValueHelper.Instance.GetStringValue(node, semanticModel);
+        public static string StringValue(this SyntaxNode node, SemanticModel semanticModel)
+        {
+            if (node != null)
+            {
+                if (node.IsAnyKind(SyntaxKind.StringLiteralExpression, SyntaxKindEx.Utf8StringLiteralExpression)
+                    && node is LiteralExpressionSyntax literal)
+                {
+                    return literal.Token.ValueText;
+                }
+                else if (node is InterpolatedStringExpressionSyntax interpolatedExpression)
+                {
+                    interpolatedExpression.TryGetGetInterpolatedTextValue(semanticModel, out var interpolatedValue);
+                    return interpolatedValue ?? interpolatedExpression.GetContentsText() ?? null;
+                }
+            }
+            return null;
+        }
```

Comment:
After addressing other comments, this could be arrow method with switch expression